spark/v4.0/build.gradle

```diff
@@ -20,6 +20,13 @@
 String sparkMajorVersion = '4.0'
 String scalaVersion = '2.13'
 
+def currentJavaVersion = JavaVersion.current()
+
+if (!(currentJavaVersion.is(JavaVersion.VERSION_17) || currentJavaVersion.is(JavaVersion.VERSION_21))) {
```

Comment:
to be more in line with other places in the code, could you please update this to 
```
JavaVersion javaVersion = JavaVersion.current()
if (javaVersion != JavaVersion.VERSION_17 && javaVersion != JavaVersion.VERSION_21) {
    throw new GradleException("Spark 4.0 build requires JDK 17 or 21 but was executed with JDK " + javaVersion)
}
```